subgraphs/euler-finance/src/mappings/handlers.ts

```diff
@@ -1,170 +1,316 @@
+  const assetStatus = getOrCreateAssetStatus(underlying);
+  const market = getOrCreateMarket(assetStatus.eToken!);
+  market.maximumLTV = event.params.newConfig.collateralFactor.toBigDecimal().div(CONFIG_FACTOR_SCALE);
+  market.liquidationThreshold = event.params.newConfig.borrowFactor.toBigDecimal().div(CONFIG_FACTOR_SCALE);
+  if (market.maximumLTV.gt(BIGDECIMAL_ZERO)) {
+    market.canUseAsCollateral = true;
+  }
+  market.save();
 }
 
 export function handleMarketActivated(event: MarketActivated): void {
-  createMarket(event);
-}
-
-function getEulerViewContract(block: ethereum.Block): EulerGeneralView {
-  const viewAddress = block.number.gt(VIEW_V2_START_BLOCK_NUMBER)
-    ? EULER_GENERAL_VIEW_V2_ADDRESS
-    : EULER_GENERAL_VIEW_ADDRESS;
-  return EulerGeneralView.bind(Address.fromString(viewAddress));
-}
+  const underlyingToken = getOrCreateToken(event.params.underlying);
+  const eToken = getOrCreateToken(event.params.eToken);
+  const dToken = getOrCreateToken(event.params.dToken);
 
-/**
- * Query Euler General View contract in order to get markets current status.
- *
- * @param marketIds List of markets to query
- * @param block current block
- * @returns query resul or null if nothing could be queried.
- */
-function queryEulerGeneralView(
-  marketIds: string[],
-  block: ethereum.Block,
-): EulerGeneralView__doQueryResultRStruct | null {
-  if (marketIds.length === 0) {
-    return null; // No market is initialized, nothing to do.
-  }
+  const market = getOrCreateMarket(eToken.id);
+  market.createdTimestamp = event.block.timestamp;
+  market.createdBlockNumber = event.block.number;
 
-  const marketAddresses: Array<Address> = marketIds.map<Address>((market: string) => Address.fromString(market));
+  // Market are initialized in isolated tier, which means currency can't be used as collateral.
+  // https://docs.euler.finance/risk-framework/tiers
+  // for borrowIsolated tier assetConfig({borrowIsolated: true})
+  // for cross tier assetConfig({borrowIsolated: false, collateralFactor: 0})
+  // for collateral tier assetConfig({borrowIsolated: false, collateralFactor: > 0})
+  market.canUseAsCollateral = false; //initial collateralFactor=0, reset in handleGovSetAssetConfig
+  market.canBorrowFrom = true;
 
-  const queryParameters: Array<ethereum.Value> = [
-    ethereum.Value.fromAddress(Address.fromString(EULER_ADDRESS)),
-    ethereum.Value.fromAddress(Address.fromString(ZERO_ADDRESS)),
-    ethereum.Value.fromAddressArray(marketAddresses),
-  ];
+  market.name = eToken.name;
+  market.inputToken = underlyingToken.id;
+  market.outputToken = eToken.id;
+  market._dToken = dToken.id;
+  market.save();
 
-  const queryParametersTuple = changetype<EulerGeneralView__doQueryInputQStruct>(queryParameters);
-  const eulerGeneralView = getEulerViewContract(block);
-  const result = eulerGeneralView.try_doQuery(queryParametersTuple);
-
-  if (result.reverted) {
-    return null;
-  }
-
-  return result.value;
+  //TODO: pToken?
```

Comment:
if you want to leave this in, you should have more context so someone else could pick this up with enough context

Reply:
I believe there is no need to handle pToken in our subgraph. Deleted.